### DISCUSSION TITLE
Issue with Deleting Links When Creating New Ones via Clicking Ports

### DISCUSSION THREAD

**Discussion by CodeSippr** [clientIO/joint#2512] [Q&A] [answered]
### Introduction

Hello JointJS Community,

I am experiencing a peculiar issue with my graph that uses logic shapes with input and output ports. I have observed that whenever I create a new link (wire) by clicking on these ports, an existing link that was created earlier through the same method gets deleted unexpectedly. However, this issue does not occur when I create or import a link using a different method. In those cases, the links are not deleted, and I am able to create multiple links without any problems.

Has anyone else encountered this issue or have any insights into what might be causing this behavior? Any suggestions or guidance would be greatly appreciated.

Thank you in advance for your help!

### Does your question relate to JointJS or JointJS+. Select both if applicable.

JointJS

**Comment by kumilingus** [comment, marked as answer]
You are probably creating a new link in `paper.options.defaultLink` with an `id` that already exists.
Adding a cell to the graph with an existing `id` fails silently. So when you start dragging a port, you are modifying the `target` of an existing link.

**Reply by CodeSippr** [reply]
That was the problem. Thank you.
